src/client/cockpit-client

```diff
@@ -19,35 +19,49 @@
 
 import gi
 import logging
+import os
 import sys
 
 gi.require_version("Gtk", "3.0")  # NOQA
 gi.require_version("WebKit2", "4.0")  # NOQA
 
 from gi.repository import GLib, Gio, Gtk, WebKit2
 
+libexecdir = os.path.realpath(__file__ + '/..')
 
+
+@Gtk.Template(filename=f'{libexecdir}/cockpit-client.ui')
 class CockpitClientWindow(Gtk.ApplicationWindow):
+    __gtype_name__ = 'CockpitClientWindow'
+    webview = Gtk.Template.Child()
+
     def __init__(self, app, uri):
         super().__init__(application=app)
 
-        self.set_size_request(800, 600)
+        self.add_action_entries([
+            ('go-back', self.go_back),
+            ('go-forward', self.go_forward),
+            ('zoom', self.zoom, 's'),
+            ('open-inspector', self.open_inspector),
+        ])
 
-        self.webview = WebKit2.WebView(is_ephemeral=True)
         self.webview.get_settings().set_enable_developer_extras(True)
         self.webview.connect('decide-policy', self.decide_policy)
-        self.add(self.webview)
-        self.webview.show()
-
         self.webview.load_uri(uri)
-        self.set_title("Loading...")  # like login.html
-        self.webview.bind_property("title", self, "title")
```

Comment:
For cockpit-desktop we need this back and hide the GtkHeaderBar, so that it continues to look like a normal desktop app (not half a web browser), at least if we are showing a particular frame (`cockpit-desktop subscriptions`) instead of the whole shell (`/`). You can keep the accelerators, I think.

So these two should hopefully not be too intrusive to conditionalize?